frontend/src/_hooks/use-mount.jsx

```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+export const useMounted = () => {
```

Comment:
Hey @arpitnath , could you explain how this `useMounted` hook ensures that the child widgets are dropped only once?

To be more specific, how is this different from using `useEffect` with an empty dependency array?

Reply:
We can only add new children once the `appDefinations` are updated. as during the first render cycle, the component (listview) mounts but the `subcontainer` mounts after it and the appDefinations is updated as soon as the widget is dropped.
As the parent mounts first and the promise is waiting to get resolved and update the state(editor), right away if we check the children which will be an empty array but on adding the children to the boxes(subcontainer layer), the app breaks. 
Here we are trying to compute components's state which is waiting to be resolved and updated resulting in accessors as undefined (component.component.name and component.component.exposedVariables)

p.s. It took me quite some time to figure out these co-relations.